core/src/main/java/org/kohsuke/stapler/export/ExportConfig.java

```diff
@@ -1,66 +1,146 @@
+        this.skipIfFail = skipIfFail;
+    }
 
     /**
-     * Interceptor used to check if given model object or it's property should be included in serialization
+     * If true, output will be indented to make it easier for humans to understand.
      */
-    private ExportInterceptor exportInterceptor = ExportInterceptor.DEFAULT;
-
-    private boolean skipIfFail = false;
-
     public boolean isPrettyPrint() {
         return prettyPrint;
     }
 
     /**
-     * If true, output will be indented to make it easier for humans to understand.
+     * Controls the behaviour of the class attribute to be produced.
      */
-    public ExportConfig withPrettyPrint(boolean prettyPrint) {
-        this.prettyPrint = prettyPrint;
-        return this;
-    }
-
     public ClassAttributeBehaviour getClassAttribute() {
         return classAttribute;
     }
 
     /**
      * Controls the behaviour of the class attribute to be produced.
      */
-    public ExportConfig withClassAttribute(ClassAttributeBehaviour cab) {
-        if (cab==null)  throw new NullPointerException();
-        this.classAttribute = cab;
-        return this;
+    public ClassAttributeBehaviour getClassAttributeBehaviour(){
+        return this.classAttribute;
     }
 
+    /**
+     * Gives {@link ExportInterceptor}. Always non-null.
+     */
     public ExportInterceptor getExportInterceptor() {
         return exportInterceptor;
     }
 
-    public ExportConfig withExportInterceptor(ExportInterceptor interceptor){
-        this.exportInterceptor = interceptor;
-        return this;
+    /**
+     *  Tells whether to skip serialization failure.
+     */
+    public boolean isSkipIfFail() {
+        return skipIfFail;
     }
 
-    public ExportConfig withSkipIfFail(boolean skipIfFail){
-        this.skipIfFail = skipIfFail;
-        return this;
+    /**
+     * Gives {@link Flavor}. Always non-null.
+     */
+    public Flavor getFlavor() {
+        return flavor;
     }
 
-    public boolean isSkipIfFail() {
-        return skipIfFail;
+    /**
+     * {@link ExportConfig} builder
+     */
+    public static class Builder{
+        private  boolean prettyPrint;
+
+        private  ClassAttributeBehaviour classAttribute = ClassAttributeBehaviour.IF_NEEDED;
+
+        private  ExportInterceptor exportInterceptor = ExportInterceptor.DEFAULT;
+
+        private  boolean skipIfFail = false;
+
+        private  final Flavor flavor;
+
+        public Builder(Flavor flavor){
+            this.flavor = flavor;
+        }
+
+        /**
+         * Turn on or off pretty printing of serialized data.
+         *
+         * @param prettyPrint default false.
+         *
+         * @return {@link Builder} instance
+         */
+        public Builder prettyPrint(boolean prettyPrint){
+            this.prettyPrint = prettyPrint;
+            return this;
+        }
+
+        /**
+         * If true serialization error will be ignored
```

Comment:
> drop this flag?

You are not using it anyway?

> Maybe we can enhance `ExportInterceptor` to handle top level bean serialization but it can come later on

Perhaps so.

Would still like a look from someone else, ideally @kohsuke.